lib/ASRouter.jsm

```diff
@@ -136,6 +142,38 @@ class _ASRouter {
     this.onMessage = this.onMessage.bind(this);
   }
 
+  _addASRouterPrefListener() {
+    if (this._prefs === null) {
+      this._prefs = Services.prefs.getBranch(ASROUTER_PREF_BRANCH);
```

Comment:
Personally, I wouldn't use a pref branch here so we don't limit ourselves to what kinds of prefs can be used for endpoints; just use Services.prefs directly